### PR TITLE
Trip model yjj

### PR DESCRIPTION
**Pull request by seoulstice**
spent way too much time on figuring out date_with_most_rides_trip_count 

I tried to modify the trip_spec so I could get the tests for these methods to pass but got an error initializing the Station, something about latitude?  I noticed that the validations for latitude/longitude are now on the station model so I tried to add those attributes in the trip_spec but was still getting those errors.  Leaving the trip_spec alone for now until tomorrow.

### CODE REVIEW

**Review by Ecksi — APPROVED** [review]
I've been running into the same issue with latitude. We'll get it sorted out today.
Looks good

**Comment by Maxscores** [comment]
Whoops, sorry about screwing everything up with the lat/long stuff. Got a little too excited about doing that part.

File: app/models/trip.rb [hunk 1]
```diff
@@ -29,6 +28,9 @@ def end_station_longitude
   scope :longest_ride, -> {maximum(:duration)}
   scope :shortest_ride, -> {minimum(:duration)}
   scope :average_ride_length, -> {average(:duration).to_f.round(2)}
-
+  scope :most_popular_starting_station, -> {maximum(:start_station)}
+  scope :most_popular_ending_station, -> {maximum(:end_station)}
+  scope :date_with_most_rides, -> {group(:start_date).order('count(*) DESC').count.first.first}
+  scope :date_with_most_rides_trip_count, -> {group(:start_date).order('count(*) DESC').count.first.last}
```
Comment (Maxscores) [inline app/models/trip.rb:34]: Definitely want to review the .first.first thing. Just seems like a red flag that there is a better way to do this.

**Comment by Maxscores** [comment]
If you pull up the current master things should be fixed on the rspec stuff. I dropped the ball there.

File: app/models/trip.rb (hunk 1 above)
Comment (icorson3) [reply app/models/trip.rb:34]: This isnt really how I want to use a scope, @Maxscores. ` The method is intended to return an ActiveRecord::Relation object, which is composable with other scopes` per the docs...so if i am returning a number, i am not properly using a scope. lets talk about it tomorrow

Comment (Ecksi) [reply app/models/trip.rb:34]: @maxscores

**Pull request merged**